services/wfm/src/main/java/org/openkilda/wfm/topology/ping/bolt/GroupCollector.java

```diff
@@ -80,7 +89,9 @@ private void collect(Tuple input) throws AbstractException {
     private CollectorDescriptor saveCurrentRecord(Tuple input) throws AbstractException {
         final PingContext pingContext = pullPingContext(input);
 
-        long expireAt = System.currentTimeMillis() + expireDelay;
+        // expiring is only a memory leakage protection in this place
+        // all timeout responses must be fired in upstream
```

Comment:
Timeout error produced by `TimeoutManager`. This bolt is responsible for collectiong ping pairs (one flow ping NB-request produce 2 pings - one for each flow direction).

If timeout is reached, `TimeoutManager` fill error field in pingContext object and send it to results processors. If we do not emit this timeout's or lost them somewhere, we should check this execution branch.

PS What is the reason to change this expiration time?

Reply:
What if on-demand ping timeout is bigger than expireDelay constant? `TimeoutManager` doesnt emit timeout yet, but in `GroupCollector` it's allready expired and dropped
Updated formula is: expireAt = current time + ping context delay ( allow `TimeoutManager` emit timeouts) + some delay for internal storm processing (for tick tuples emitting and bolt communication)